astropy/units/quantity.py

```diff
@@ -2235,3 +2237,13 @@ def _unquantify_allclose_arguments(actual, desired, rtol, atol):
         raise UnitsError("'rtol' should be dimensionless")
 
     return actual.value, desired.value, rtol.value, atol.value
+
+
+_QuantityT = TypeVar("_QuantityT", bound=Quantity)
+
+
+@replace.register
+def _replace_quantity(obj: _QuantityT, /, **kwargs: Any) -> _QuantityT:
```

Comment:
This handles Quantity subclasses, but we might want some more specific functions that better address nuances.